src/mpid/ch4/src/ch4r_rma.h

```diff
@@ -705,15 +718,25 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_CH4U_mpi_raccumulate(const void *origin_addr,
                                                         MPIR_Request ** request)
 {
     int mpi_errno = MPI_SUCCESS;
+    MPIR_Request *sreq = NULL;
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_CH4U_MPI_RACCUMULATE);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_CH4U_MPI_RACCUMULATE);
 
     mpi_errno = MPIDI_do_accumulate(origin_addr, origin_count, origin_datatype,
                                     target_rank, target_disp, target_count, target_datatype, op,
-                                    win, request);
+                                    win, &sreq);
     if (mpi_errno)
         MPIR_ERR_POP(mpi_errno);
+
+    if (sreq == NULL) {
+        /* create a completed request for user if issuing is completed immediately. */
+        sreq = MPIR_Request_create(MPIR_REQUEST_KIND__RMA);
+        MPIR_Assert(sreq);
+        MPIDI_Request_set_completed(sreq);
```

Comment:
Can you use MPIDI_Request_complete instead of set_completed?  They should be equivalent in this case since the cc is 1.  But I'd prefer to never have direct setting of values, and only increment/decrement as needed.